File: src/content/docs/topics/cs/coding-problems/two-pointers/443-string-compression-approach2.ts
function compress(chars: string[]): number {
    let write = 0;
    let i = 0;
    while (i < chars.length) {
        const char = chars[i];
        let count = 0;
        while (i < chars.length && chars[i] === char) {
            count++;
            i++;
        }
        chars[write++] = char;
        if (count > 1) {
            for (const digit of String(count)) {
                chars[write++] = digit;
            }
        }
    }
    return write;
}

const chars1 = ['a', 'a', 'b', 'b', 'c', 'c', 'c'];
const n1 = compress(chars1);
console.log(n1, chars1.slice(0, n1));  // 6 ['a','2','b','2','c','3']

File: src/content/docs/topics/cs/coding-problems/two-pointers/443-string-compression-approach2.py
from typing import List

def compress(chars: List[str]) -> int:
    write = 0
    i = 0
    while i < len(chars):
        char = chars[i]
        count = 0
        while i < len(chars) and chars[i] == char:
            count += 1
            i += 1
        chars[write] = char
        write += 1
        if count > 1:
            for digit in str(count):
                chars[write] = digit
                write += 1
    return write

chars1 = ['a', 'a', 'b', 'b', 'c', 'c', 'c']
n1 = compress(chars1)
print(n1, chars1[:n1])  # 6 ['a', '2', 'b', '2', 'c', '3']
